assets/js/components/KeyMetrics/ConfirmSitePurposeChangeModal.js

```diff
@@ -48,15 +48,23 @@ import {
 	USER_INPUT_QUESTIONS_PURPOSE,
 } from '../user-input/util/constants';
 import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
 
 function ConfirmSitePurposeChangeModal( {
 	dialogActive = false,
 	handleDialog = null,
 } ) {
 	const [ isSaving, setIsSaving ] = useState( false );
 
+	const includeConversionTailoredMetrics = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).haveConversionEventsForTailoredMetrics()
+	);
```

Comment:
And here as well, we need to ensure that the Analytics module is connected before calling its selector.